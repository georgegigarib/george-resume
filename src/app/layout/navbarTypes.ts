export interface boxType {
  startRow: number
  spanRow: number
  startColumn: number
  spanColumn: number
  content: string
}

export enum Sections {
  Dev = '#dev',
  Me = '#me',
  Music = '#music'
}
