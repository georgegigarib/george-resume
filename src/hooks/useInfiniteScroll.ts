import { useEffect, useRef } from 'react'

const useInfiniteScroll = (containerId: string) => {
  const containerRef = useRef<HTMLDivElement>(null)

  useEffect(() => {
    const container = containerRef.current
    if (!container) return

    let animationFrame: number
    let scrollPosition = 0

    try {
      // Duplicamos el contenido para permitir un scroll infinito
      const originalContent = container.innerHTML
      container.innerHTML += originalContent

      // Calculamos el ancho total para ajustar el desplazamiento
      const scrollWidth = container.scrollWidth / 2

      const scrollStep = () => {
        scrollPosition += 0.5 // Ajusta la velocidad del desplazamiento aquÃ­
        if (scrollPosition >= scrollWidth) {
          scrollPosition = 0
        }
        container.scrollLeft = scrollPosition
        animationFrame = requestAnimationFrame(scrollStep)
      }

      scrollStep()
    } catch (error) {
      console.error('Error in useInfiniteScroll:', error)
    }

    // Limpieza al desmontar
    return () => {
      if (animationFrame) cancelAnimationFrame(animationFrame)
    }
  }, [containerId])

  return containerRef
}

export default useInfiniteScroll
