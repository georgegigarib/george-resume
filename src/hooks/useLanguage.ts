import { useTranslation } from 'react-i18next';

export const useLanguage = () => {
  const { i18n } = useTranslation();

  const changeLanguage = (lng: string) => {
    i18n.changeLanguage(lng);
    localStorage.setItem('language', lng);
  };

  const loadLanguageFromStorage = () => {
    const savedLanguage = localStorage.getItem('language') || 'en';
    i18n.changeLanguage(savedLanguage);
  };

  return { changeLanguage, loadLanguageFromStorage };
};
