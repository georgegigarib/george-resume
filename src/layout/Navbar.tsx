import { useState, useEffect } from "react";
import { useLocation, useNavigate } from "react-router-dom";

const Navbar = () => {
  const [activeHash, setActiveHash] = useState('');
  const [translateClass, setTranslateClass] = useState("");
  const location = useLocation();
  const navigate = useNavigate();

  const scrollToSection = (id: string) => {
    const section = document.querySelector(id) as HTMLElement;
    if (section) {
      const offset =
        section.offsetLeft - window.innerWidth / 2 + section.clientWidth / 2;

      window.scrollTo({
        left: offset,
      });

      const newHash = `${id}`;
      navigate(`${location.pathname}${newHash}`, { replace: true });
    }
  };

  useEffect(() => {
    const hash = location.hash || "#me";
    setActiveHash(hash);

    const getTranslateXForHash = (hash: string) => {
      switch (hash) {
        case "#dev":
          return "translate-x-[48%] xl:translate-x-[52%]";
        case "#me":
          return "-translate-x-0";
        case "#music":
          return "-translate-x-[44%] sm:-translate-x-[40%] md:-translate-x-[45%] xl:-translate-x-[50%]"; 
        default:
          return "translate-x-0";
      }
    };

    setTranslateClass(getTranslateXForHash(hash));
    scrollToSection(hash)
  }, [location.hash]);

  return (
    <nav className="bg-transparent border-transparent  fixed w-full z-20 top-0 start-0 border-b   nav-bottom-oval h-20">
      <div className="max-w-screen-xl flex flex-wrap items-center justify-center mx-auto p-8">
        <div
          className={`justify-between flex mt-2 h-16 xs:gap-28 sm:gap-56 md:gap-80 lg:gap-[450px] xl:gap-[600px] transform ${translateClass}  animated-transform`}
          id="navbar-sticky"
        >
          <a
            onClick={() => scrollToSection("#dev")}
            className={`block rounded md:p-0 ${
              activeHash === "#dev"
                ? "text-white font-bold animated-transform translate-y-[45%] md:translate-y-[45%] scale-130"
                : "text-gray-900 animated-transform rotate-[18deg] translate-y-[10%] sm:rotate-12 md:translate-y-[0%] xl:rotate-3 opacity-40 font-bold s 2xl:translate-y-[20%]"
            } ${activeHash === "#music" ? "-translate-y-[60%] md:-translate-y-[80%] 2xl:-translate-y-[50%]" : ""}`}
            aria-current={activeHash === "#dev" ? "page" : undefined}
          >
            Dev
          </a>
          <a
            onClick={() => scrollToSection("#me")}
            className={`block pt-8 rounded md:p-0 ${
              activeHash === "#me"
                ? "text-white font-bold animated-transform -translate-y-[20%] md:translate-y-[50%] scale-130"
                : "text-gray-900 -translate-y-[40%] sm:-translate-y-[40%] animated-transform md:dark:hover:bg-transparent opacity-40 font-bold"
            } ${
              activeHash === "#music"
                ? "rotate-[18deg]  sm:rotate-6 md:-translate-y-[0%] 2xl:translate-y-[15%] lg:rotate-3"
                : activeHash === "#dev"
                ? "-rotate-[18deg] sm:-rotate-6  md:translate-y-[0%]  2xl:translate-y-[15%] lg:-rotate-3"
                : ""
            }`}
          >
            Me
          </a>
          <a
            onClick={() => scrollToSection("#music")}
            className={`block rounded md:p-0 ${
              activeHash === "#music"
                ? "text-white font-bold animated-transform translate-y-[45%] scale-130 -rotate-1"
                : "text-gray-900 -rotate-[18deg] translate-y-[10%] sm:-rotate-12 md:translate-y-[0%] xl:-rotate-3 animated-transform opacity-40 font-bold 2xl:translate-y-[20%]"
            } ${activeHash === "#dev" ? "-translate-y-[60%] md:-translate-y-[80%] 2xl:-translate-y-[50%]" : ""}`}
          >
            Music
          </a>
        </div>
      </div>
    </nav>
  );
};

export default Navbar;
