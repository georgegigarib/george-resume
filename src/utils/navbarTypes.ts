export interface boxType {
    startRow: number,
    spanRow: number,
    startColumn: number,
    spanColumn: number,
    content: string
}