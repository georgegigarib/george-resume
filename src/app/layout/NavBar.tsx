import { useEffect, useState } from 'react';
import { useDispatch, useSelector } from 'react-redux';
import { useLocation } from 'react-router-dom';
import { setActiveHash } from '@/store/hash/activeHashSlice';
import LanguageSwitcher from '@/components/organisms/NavBarDropdownMenu';
import { Sections } from '@/app/layout/navbarTypes';
import { RootState } from "@/store/hash/activeHashSlice";

import DevLink from '@/components/atoms/DevLink';
import MeLink from '@/components/atoms/MeLink.tsx';
import MusicLink from '@/components/atoms/MusicLink';

const Navbar = () => {
  const dispatch = useDispatch();
  const location = useLocation();
  const activeHash = useSelector((state: RootState) => state.hash.activeHash);
  const [translateClass, setTranslateClass] = useState('');
  const [automaticScroll, setAutomaticScroll] = useState(true);

  const moveNavBar = (hash: string): void => {
    const getTranslateXForHash = (hash: string) => {
      switch (hash) {
        case Sections.Dev:
          return 'translate-x-[48%] xl:translate-x-[52%]';
        case Sections.Me:
          return '-translate-x-0';
        case Sections.Music:
          return '-translate-x-[44%] sm:-translate-x-[45%] md:-translate-x-[45%] xl:-translate-x-[50%]';
        default:
          return 'translate-x-0';
      }
    };

    setTranslateClass(getTranslateXForHash(hash));
  };

  const scrollToSection = (id: string, called?: boolean) => {
    const section = document.querySelector(id) as HTMLElement;
    if (section) {
      const offset = section.offsetLeft - window.innerWidth / 2 + section.clientWidth / 2;
      setAutomaticScroll(true);

      window.scrollTo({
        left: offset,
        behavior: 'smooth',
      });

      setTimeout(() => {
        setAutomaticScroll(false);
      }, called ? 0 : 500);
    }
  };

  useEffect(() => {
    const hash = location.hash || Sections.Me;
    dispatch(setActiveHash(hash)); // Despacha la acción para establecer el hash activo
    moveNavBar(hash);

    if (automaticScroll) {
      scrollToSection(hash);
      return;
    }
  }, [location.hash, automaticScroll, dispatch]); // Asegúrate de incluir dispatch como dependencia

  useEffect(() => {
    if (automaticScroll) return;

    const handleScroll = () => {
      const scrollX = window.scrollX;
      const windowWidth = window.innerWidth;
      const documentWidth = document.body.scrollWidth;

      const scrollPercentage = (scrollX / (documentWidth - windowWidth)) * 100;

      if (scrollPercentage > 66) {
        moveNavBar(Sections.Music);
        dispatch(setActiveHash(Sections.Music));
        history.pushState(null, '', Sections.Music);
      } else if (scrollPercentage < 33) {
        moveNavBar(Sections.Dev);
        dispatch(setActiveHash(Sections.Dev));
        history.pushState(null, '', Sections.Dev);
      } else {
        moveNavBar(Sections.Me);
        dispatch(setActiveHash(Sections.Me));
        history.pushState(null, '', Sections.Me);
      }
    };

    window.addEventListener('scroll', handleScroll);
    return () => {
      window.removeEventListener('scroll', handleScroll);
    };
  }, [automaticScroll, dispatch]);

  return (
    <nav className="bg-transparent border-transparent fixed w-full z-20 top-0 start-0 border-b nav-bottom-oval h-12 md:h-20">
      <LanguageSwitcher />

      <div className="max-w-screen-xl flex flex-wrap items-center justify-center mx-auto p-8">
        <div
          className={`justify-between flex mt-0 md:mt-2 h-16 gap-20 xs:gap-28 sm:gap-56 md:gap-80 lg:gap-[450px] xl:gap-[600px] transform ${translateClass} animated-transform`}
          id="navbar-sticky"
        >
          <DevLink scrollToSection={scrollToSection} setAutomaticScroll={setAutomaticScroll} activeHash={activeHash} />
          <MeLink scrollToSection={scrollToSection} setAutomaticScroll={setAutomaticScroll} activeHash={activeHash} />
          <MusicLink scrollToSection={scrollToSection} setAutomaticScroll={setAutomaticScroll} activeHash={activeHash} />
        </div>
      </div>
    </nav>
  );
};

export default Navbar;
