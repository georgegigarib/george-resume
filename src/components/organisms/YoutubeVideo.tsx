import React, { useRef, useState, useEffect } from 'react'
import VideoSrc from '@/assets/videos/guitar-video.mp4'
import { Tooltip, CircularProgress } from '@mui/material'
import { useTranslation } from 'react-i18next'
import { useSelector } from 'react-redux'
import { Sections } from '@/app/layout/types/NavBarTypes'
import { useIsMobile } from '@/hooks/useIsMobile'
import { StoreState } from '@/store'
import VolumeOffIcon from '@mui/icons-material/VolumeOff'
import VolumeUpIcon from '@mui/icons-material/VolumeUp'

export default function YoutubeVideo(): React.ReactElement {
  const videoUrl = 'https://www.youtube.com/watch?v=Mfxv1jmkWlk'
  const isMobile = useIsMobile()
  const videoRef = useRef<HTMLVideoElement>(null)
  const [isMuted, setIsMuted] = useState(true)
  const [showTooltip, setShowTooltip] = useState(false)
  const [isLoading, setIsLoading] = useState(true)
  const activeHash = useSelector((state: StoreState) => state.hash.activeHash)
  const { t } = useTranslation()

  const handleVideoClick = (): void => {
    if (videoRef.current) {
      const mutedState = !isMuted
      videoRef.current.muted = mutedState
      setIsMuted(mutedState)
    }
  }

  useEffect(() => {
    const timer = setTimeout(
      () => {
        setShowTooltip(activeHash === Sections.Music)
      },
      activeHash === Sections.Music ? 3500 : 1500
    )

    return () => clearTimeout(timer)
  }, [activeHash])

  const handleVideoLoaded = (): void => {
    setIsLoading(false)
  }

  return (
    <div className="relative flex flex-col justify-between rounded-xl h-full">
      {isLoading && (
        <div className="absolute inset-0 flex justify-center items-center bg-gray-700 rounded-xl z-10">
          <CircularProgress className="text-white" />
        </div>
      )}

      <video
        ref={videoRef}
        className={`rounded-xl cursor-pointer ${isLoading ? 'opacity-0' : 'opacity-100'}`}
        width={isMobile ? '110px' : '210px'}
        autoPlay
        loop
        muted={isMuted}
        onClick={handleVideoClick}
        onLoadedData={handleVideoLoaded}
        title="Click to mute/unmute"
        playsInline
      >
        <source src={VideoSrc} type="video/mp4" />
      </video>

      <div className="absolute bottom-3 left-2">
        {isMuted ? <VolumeOffIcon className="text-white text-2xl" /> : <VolumeUpIcon className="text-white text-2xl" />}
      </div>

      <div
        className={`absolute -bottom-7 left-0 right-0 -z-10 transition-transform duration-1000 ease-in-out transform ${
          showTooltip ? 'translate-y-0 opacity-100' : '-translate-y-10 opacity-0'
        }`}
      >
        <Tooltip title={t('music.video.seeFullVideo')}>
          <span className="cursor-pointer w-full">
            <a
              href={videoUrl}
              target="_blank"
              className="bg-gray-700 text-white w-full pt-4 pb-1 text-sm font-bold rounded-b-lg opacity-75 hover:opacity-80 transition-opacity block text-center"
            >
              {t('music.video.clickMe')}
            </a>
          </span>
        </Tooltip>
      </div>
    </div>
  )
}
