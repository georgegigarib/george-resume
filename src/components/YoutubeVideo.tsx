import React, { useRef, useState, useEffect } from 'react';
import { useIsMobile } from '../hooks/useIsMobile';
import VolumeOffIcon from '@mui/icons-material/VolumeOff';
import VolumeUpIcon from '@mui/icons-material/VolumeUp';
import VideoSrc from './../assets/final.mp4';
import { Tooltip } from '@mui/material';
import { useTranslation } from 'react-i18next';
import { Sections } from '../utils/navbarTypes';
import { useSelector } from 'react-redux';
import { RootState } from '../store/types';

export default function LocalVideoWithTooltip(): React.ReactElement {
  const isMobile = useIsMobile();
  const videoRef = useRef<HTMLVideoElement>(null);
  const [isMuted, setIsMuted] = useState(true);
  const [showTooltip, setShowTooltip] = useState(false);
  const activeHash = useSelector((state: RootState) => state.hash.activeHash);
  const { t } = useTranslation();

  const handleVideoClick = (): void => {
    if (videoRef.current) {
      const mutedState = !isMuted;
      videoRef.current.muted = mutedState;
      setIsMuted(mutedState);
    }
  };

  useEffect(() => {
    const timer = setTimeout(() => {
      setShowTooltip(activeHash === Sections.Music);
    }, activeHash === Sections.Music ? 3500 : 1500);

    return () => clearTimeout(timer);
  }, [activeHash]);

  const handleTooltipClick = (): void => {
    window.open('https://www.youtube.com/watch?v=Mfxv1jmkWlk', '_blank');
  };

  return (
    <div className="relative flex flex-col justify-between rounded-xl bg-gray-700">
      <video
        ref={videoRef}
        className="rounded-xl cursor-pointer"
        width={isMobile ? '110px' : '200px'}
        autoPlay
        loop
        muted={isMuted}
        onClick={handleVideoClick}
        title="Click to mute/unmute"
        playsInline
      >
        <source src={VideoSrc} type="video/mp4" />
      </video>

      <div className="absolute bottom-3 left-2">
        {isMuted ? <VolumeOffIcon className="text-white text-2xl" /> : <VolumeUpIcon className="text-white text-2xl" />}
      </div>

      <div
        className={`absolute -bottom-7 left-0 right-0 -z-10 transition-transform duration-1000 ease-in-out transform ${
          showTooltip ? 'translate-y-0 opacity-100' : '-translate-y-10 opacity-0'
        }`}
      >
        <Tooltip title={t('music.video.seeFullVideo')}>
          <span onClick={handleTooltipClick} className="cursor-pointer w-full"> 
            <a className="bg-gray-700 text-white w-full pt-4 pb-1 text-sm font-bold rounded-b-lg opacity-75 hover:opacity-80 transition-opacity block text-center">
              {t('music.video.clickMe')}
            </a>
          </span>
        </Tooltip>
      </div>
    </div>
  );
}
