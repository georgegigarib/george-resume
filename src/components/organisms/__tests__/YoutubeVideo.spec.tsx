import { render, screen, waitFor } from '@testing-library/react'
import { describe, it, expect, vi, Mock } from 'vitest'
import { useIsMobile } from '@/hooks/useIsMobile'
import { Provider } from 'react-redux'
import { configureStore, Store } from '@reduxjs/toolkit'
import activeHashReducer from '@/store/hash/activeHashSlice'
import { Sections } from '@/app/layout/navbarTypes'
import YoutubeVideo from '../YoutubeVideo'
import i18n from '@/locales/i18n.config'
import { I18nextProvider } from 'react-i18next'

const mockStore = (initialState: { hash: { activeHash: Sections } }) => {
  return configureStore({
    reducer: {
      hash: activeHashReducer
    },
    preloadedState: initialState
  })
}

vi.mock('@/hooks/useIsMobile', () => ({
  useIsMobile: vi.fn()
}))

describe('LocalVideoWithTooltip Component', () => {
  const renderComponent = (store: Store) =>
    render(
      <I18nextProvider i18n={i18n}>
        <Provider store={store}>
          <YoutubeVideo />
        </Provider>
      </I18nextProvider>
    )

  it('renders correctly and displays loading indicator initially', () => {
    ;(useIsMobile as Mock).mockReturnValue(false)
    const store = mockStore({ hash: { activeHash: Sections.Dev } })

    renderComponent(store)

    expect(screen.getByRole('progressbar')).toBeVisible()
  })

  it('matches snapshot', () => {
    ;(useIsMobile as Mock).mockReturnValue(false)
    const store = mockStore({ hash: { activeHash: Sections.Dev } })

    const { container } = renderComponent(store)
    expect(container).toMatchSnapshot()
  })

  it('shows tooltip when activeHash matches and after timeout', async () => {
    ;(useIsMobile as Mock).mockReturnValue(false)
    const store = mockStore({ hash: { activeHash: Sections.Music } })
    renderComponent(store)

    await waitFor(() => expect(screen.getByText('Click me!')).toBeVisible())
  })
})
