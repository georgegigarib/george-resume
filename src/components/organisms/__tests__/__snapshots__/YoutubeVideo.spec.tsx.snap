// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`LocalVideoWithTooltip Component > matches snapshot 1`] = `
<div>
  <div
    class="relative flex flex-col justify-between rounded-xl h-full bg-gray-700"
  >
    <div
      class="absolute inset-0 flex justify-center items-center bg-gray-700 rounded-xl z-10"
    >
      <span
        class="MuiCircularProgress-root MuiCircularProgress-indeterminate MuiCircularProgress-colorPrimary text-white css-1yonwsg-MuiCircularProgress-root"
        role="progressbar"
        style="width: 40px; height: 40px;"
      >
        <svg
          class="MuiCircularProgress-svg css-1idz92c-MuiCircularProgress-svg"
          viewBox="22 22 44 44"
        >
          <circle
            class="MuiCircularProgress-circle MuiCircularProgress-circleIndeterminate css-1a8epmm-MuiCircularProgress-circle"
            cx="44"
            cy="44"
            fill="none"
            r="20.2"
            stroke-width="3.6"
          />
        </svg>
      </span>
    </div>
    <video
      autoplay=""
      class="rounded-xl cursor-pointer opacity-0"
      loop=""
      playsinline=""
      title="Click to mute/unmute"
      width="200px"
    >
      <source
        src="/src/assets/videos/final.mp4"
        type="video/mp4"
      />
    </video>
    <div
      class="absolute bottom-3 left-2"
    >
      <svg
        aria-hidden="true"
        class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium text-white text-2xl css-20bmp1-MuiSvgIcon-root"
        data-testid="VolumeOffIcon"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M16.5 12c0-1.77-1.02-3.29-2.5-4.03v2.21l2.45 2.45c.03-.2.05-.41.05-.63m2.5 0c0 .94-.2 1.82-.54 2.64l1.51 1.51C20.63 14.91 21 13.5 21 12c0-4.28-2.99-7.86-7-8.77v2.06c2.89.86 5 3.54 5 6.71M4.27 3 3 4.27 7.73 9H3v6h4l5 5v-6.73l4.25 4.25c-.67.52-1.42.93-2.25 1.18v2.06c1.38-.31 2.63-.95 3.69-1.81L19.73 21 21 19.73l-9-9zM12 4 9.91 6.09 12 8.18z"
        />
      </svg>
    </div>
    <div
      class="absolute -bottom-7 left-0 right-0 -z-10 transition-transform duration-1000 ease-in-out transform -translate-y-10 opacity-0"
    >
      <span
        aria-label="See full video"
        class="cursor-pointer w-full"
        data-mui-internal-clone-element="true"
      >
        <a
          class="bg-gray-700 text-white w-full pt-4 pb-1 text-sm font-bold rounded-b-lg opacity-75 hover:opacity-80 transition-opacity block text-center"
          href="https://www.youtube.com/watch?v=Mfxv1jmkWlk"
          target="_blank"
        >
          Click me!
        </a>
      </span>
    </div>
  </div>
</div>
`;

exports[`LocalVideoWithTooltip Component > renders correctly and displays loading indicator initially 1`] = `
<DocumentFragment>
  <div
    class="relative flex flex-col justify-between rounded-xl h-full bg-gray-700"
  >
    <div
      class="absolute inset-0 flex justify-center items-center bg-gray-700 rounded-xl z-10"
    >
      <span
        class="MuiCircularProgress-root MuiCircularProgress-indeterminate MuiCircularProgress-colorPrimary text-white css-1yonwsg-MuiCircularProgress-root"
        role="progressbar"
        style="width: 40px; height: 40px;"
      >
        <svg
          class="MuiCircularProgress-svg css-1idz92c-MuiCircularProgress-svg"
          viewBox="22 22 44 44"
        >
          <circle
            class="MuiCircularProgress-circle MuiCircularProgress-circleIndeterminate css-1a8epmm-MuiCircularProgress-circle"
            cx="44"
            cy="44"
            fill="none"
            r="20.2"
            stroke-width="3.6"
          />
        </svg>
      </span>
    </div>
    <video
      autoplay=""
      class="rounded-xl cursor-pointer opacity-0"
      loop=""
      playsinline=""
      title="Click to mute/unmute"
      width="200px"
    >
      <source
        src="/src/assets/videos/final.mp4"
        type="video/mp4"
      />
    </video>
    <div
      class="absolute bottom-3 left-2"
    >
      <svg
        aria-hidden="true"
        class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium text-white text-2xl css-20bmp1-MuiSvgIcon-root"
        data-testid="VolumeOffIcon"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M16.5 12c0-1.77-1.02-3.29-2.5-4.03v2.21l2.45 2.45c.03-.2.05-.41.05-.63m2.5 0c0 .94-.2 1.82-.54 2.64l1.51 1.51C20.63 14.91 21 13.5 21 12c0-4.28-2.99-7.86-7-8.77v2.06c2.89.86 5 3.54 5 6.71M4.27 3 3 4.27 7.73 9H3v6h4l5 5v-6.73l4.25 4.25c-.67.52-1.42.93-2.25 1.18v2.06c1.38-.31 2.63-.95 3.69-1.81L19.73 21 21 19.73l-9-9zM12 4 9.91 6.09 12 8.18z"
        />
      </svg>
    </div>
    <div
      class="absolute -bottom-7 left-0 right-0 -z-10 transition-transform duration-1000 ease-in-out transform -translate-y-10 opacity-0"
    >
      <span
        aria-label="See full video"
        class="cursor-pointer w-full"
        data-mui-internal-clone-element="true"
      >
        <a
          class="bg-gray-700 text-white w-full pt-4 pb-1 text-sm font-bold rounded-b-lg opacity-75 hover:opacity-80 transition-opacity block text-center"
          href="https://www.youtube.com/watch?v=Mfxv1jmkWlk"
          target="_blank"
        >
          Click me!
        </a>
      </span>
    </div>
  </div>
</DocumentFragment>
`;

exports[`LocalVideoWithTooltip Component > shows tooltip when activeHash matches and after timeout 1`] = `
<DocumentFragment>
  <div
    class="relative flex flex-col justify-between rounded-xl h-full bg-gray-700"
  >
    <div
      class="absolute inset-0 flex justify-center items-center bg-gray-700 rounded-xl z-10"
    >
      <span
        class="MuiCircularProgress-root MuiCircularProgress-indeterminate MuiCircularProgress-colorPrimary text-white css-1yonwsg-MuiCircularProgress-root"
        role="progressbar"
        style="width: 40px; height: 40px;"
      >
        <svg
          class="MuiCircularProgress-svg css-1idz92c-MuiCircularProgress-svg"
          viewBox="22 22 44 44"
        >
          <circle
            class="MuiCircularProgress-circle MuiCircularProgress-circleIndeterminate css-1a8epmm-MuiCircularProgress-circle"
            cx="44"
            cy="44"
            fill="none"
            r="20.2"
            stroke-width="3.6"
          />
        </svg>
      </span>
    </div>
    <video
      autoplay=""
      class="rounded-xl cursor-pointer opacity-0"
      loop=""
      playsinline=""
      title="Click to mute/unmute"
      width="200px"
    >
      <source
        src="/src/assets/videos/final.mp4"
        type="video/mp4"
      />
    </video>
    <div
      class="absolute bottom-3 left-2"
    >
      <svg
        aria-hidden="true"
        class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium text-white text-2xl css-20bmp1-MuiSvgIcon-root"
        data-testid="VolumeOffIcon"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M16.5 12c0-1.77-1.02-3.29-2.5-4.03v2.21l2.45 2.45c.03-.2.05-.41.05-.63m2.5 0c0 .94-.2 1.82-.54 2.64l1.51 1.51C20.63 14.91 21 13.5 21 12c0-4.28-2.99-7.86-7-8.77v2.06c2.89.86 5 3.54 5 6.71M4.27 3 3 4.27 7.73 9H3v6h4l5 5v-6.73l4.25 4.25c-.67.52-1.42.93-2.25 1.18v2.06c1.38-.31 2.63-.95 3.69-1.81L19.73 21 21 19.73l-9-9zM12 4 9.91 6.09 12 8.18z"
        />
      </svg>
    </div>
    <div
      class="absolute -bottom-7 left-0 right-0 -z-10 transition-transform duration-1000 ease-in-out transform -translate-y-10 opacity-0"
    >
      <span
        aria-label="See full video"
        class="cursor-pointer w-full"
        data-mui-internal-clone-element="true"
      >
        <a
          class="bg-gray-700 text-white w-full pt-4 pb-1 text-sm font-bold rounded-b-lg opacity-75 hover:opacity-80 transition-opacity block text-center"
          href="https://www.youtube.com/watch?v=Mfxv1jmkWlk"
          target="_blank"
        >
          Click me!
        </a>
      </span>
    </div>
  </div>
</DocumentFragment>
`;
